compiler/src/dotty/tools/dotc/typer/Typer.scala

```diff
@@ -715,7 +715,7 @@ class Typer extends Namer
 
   def escapingRefs(block: Tree, localSyms: => List[Symbol])(implicit ctx: Context): collection.Set[NamedType] = {
     lazy val locals = localSyms.toSet
-    block.tpe namedPartsWith (tp => locals.contains(tp.symbol))
+    block.tpe.namedPartsWith(tp => locals.contains(tp.symbol) && !tp.widen.isErroneous)
```

Comment:
`isErroneous` already calls `widen`, so this could be:
```suggestion
    block.tpe.namedPartsWith(tp => locals.contains(tp.symbol) && !tp.isErroneous)
```